Numark-Mixtrack-Pro-FX-scripts.js

```diff
@@ -355,7 +375,68 @@ MixtrackProFX.Deck = function(number) {
             deck.pads.reconnectComponents();
         }
     });
+    
+    // switch pad mode to shifted sampler
+    this.modeSampleShift = new components.Button({
+        input: function(channel, control, value, status, group) {
+            midi.sendShortMsg(0x90 + channel, 0x00, 0x01); // hotcue
+            midi.sendShortMsg(0x90 + channel, 0x0D, 0x01); // auto loop
+            midi.sendShortMsg(0x90 + channel, 0x07, 0x01); // fader cuts
+            midi.sendShortMsg(0x90 + channel, 0x0B, 0x01); // sample
+            midi.sendShortMsg(0x90 + channel, 0x0F, 0x7F); // sample shifted
+            midi.sendShortMsg(0x90 + channel, 0x02, 0x01); // beatjump
+
+            for (var i = 0; i < 8; i++) {
+                deck.pads[i].group = "[Sampler" + (i + 9) + "]";
+                deck.pads[i].inKey = "cue_gotoandplay";
+                deck.pads[i].outKey = "play";
+
+                deck.padsShift[i].group = "[Sampler" + (i + 9) + "]";
+                deck.padsShift[i].inKey = "start_stop";
+            }
+
+            deck.pads.reconnectComponents();
+        }
+    });
+    
+    // switch pad mode to beatjump
+    this.modeBeatjump = new components.Button({
+        input: function(channel, control, value, status, group) {
+            midi.sendShortMsg(0x90 + channel, 0x00, 0x01); // hotcue            
+            midi.sendShortMsg(0x90 + channel, 0x0D, 0x01); // auto loop
+            midi.sendShortMsg(0x90 + channel, 0x07, 0x01); // fader cuts
+            midi.sendShortMsg(0x90 + channel, 0x0B, 0x01); // sample
+            midi.sendShortMsg(0x90 + channel, 0x0F, 0x01); // sample shifted
+            midi.sendShortMsg(0x90 + channel, 0x02, 0x7F); // beatjump
+
+            // this is just sad
+            deck.pads[0].inKey = "beatjump_0.0625_forward";
+            deck.pads[1].inKey = "beatjump_0.125_forward";
+            deck.pads[2].inKey = "beatjump_0.25_forward";
+            deck.pads[3].inKey = "beatjump_0.5_forward";
+            deck.pads[4].inKey = "beatjump_1_forward";
+            deck.pads[5].inKey = "beatjump_2_forward";
+            deck.pads[6].inKey = "beatjump_forward"; // this button is adjustable, it uses the value set in the interface (4 by default) 
```

Comment:
I think it would make more sense to use beatjump_4_forward. I would expect each pad to beatjump 2x more than previous one.